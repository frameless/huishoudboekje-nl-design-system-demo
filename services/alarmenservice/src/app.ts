import cors from "cors";
import express from "express";
import apiV1Router from "./api-v1";
import {errorHandler} from "./errorHandlers";
import healthRouter from "./health";

const app = express();
const DATABASE_URL = process.env.DATABASE_URL;

if (!DATABASE_URL && process.env.NODE_ENV !== "test") {
	throw new Error("Unknown database. Please provide the database connection string in DATABASE_URL in the environment.");
}

const logger = () => (req, res, next) => {
	console.log("[EVENT]", `Incoming request: ${req.method} ${req.url}`);
	if (req.body) {
		console.log("[EVENT]", "Body:", req.body);
	}
	next();
};

app.use(cors());
app.use(express.json());
if (process.env.NODE_ENV === 'dev') {
	app.use(logger());
}

app.get("/health", healthRouter);
app.use("/v1/alarms", apiV1Router);
app.use(errorHandler);

export default app;