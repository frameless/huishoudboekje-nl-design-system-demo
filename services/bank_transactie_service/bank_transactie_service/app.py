""" Main app for bank_transactie_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from bank_transactie_service.views.bank_transaction import BankTransactionView
from bank_transactie_service.views.customer_statement_message import CustomerStatementMessageView
from core_service import database
from bank_transactie_service.views.bank_transaction_range import BanktransactionRangeView
from bank_transactie_service.views.transactions_filter_view import BanktransactionFilterView
from bank_transactie_service.views.bank_transaction_sum import BanktransactionSumView
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='bank_transactie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    add_sqlalchemy_statsd_metrics(app)


    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/customerstatementmessages", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_view"},
        {"path": "/customerstatementmessages/<object_id>", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_detail_view"},
        {"path": "/banktransactions", "view": BankTransactionView,
            "name": "banktransaction_view"},
        {"path": "/banktransactions/<object_id>", "view": BankTransactionView,
            "name": "banktransaction_detail_view"},
        {"path": "/banktransactions/filter", "view": BanktransactionFilterView,
            "name": "banktransaction_filter_view"},
        {"path": "/banktransactions/range", "view": BanktransactionRangeView,
            "name": "banktransaction_range_view"},
        {"path": "/banktransactions/sum", "view": BanktransactionSumView,
            "name": "banktransaction_sum_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
