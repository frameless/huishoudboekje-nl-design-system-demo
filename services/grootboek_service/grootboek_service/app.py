""" Main app for grootboek_service """
from models import *
import logging
from flask import Flask, Response
from flask_migrate import Migrate
from core_service import database
from grootboek_service.views.grootboekrekeningen import GrootboekrekeningenView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(config_name='grootboek_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/grootboekrekeningen", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_view"},
        {"path": "/grootboekrekeningen/<object_id>", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_detail_view"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
