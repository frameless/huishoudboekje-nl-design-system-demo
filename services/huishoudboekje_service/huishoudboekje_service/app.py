""" Main app for huishoudboekje_service """
import logging
import os
from flask import Flask, Response
from flask_migrate import Migrate

from huishoudboekje_service.views import (
    BurgerView,
    AfspraakView,
    RekeningView,
    RekeningBurgerView,
    RekeningAfdelingView,
    JournaalpostView,
    RubriekView,
    ConfiguratieView,
    OverschrijvingView,
    ExportView,
    HuishoudenView,
    AfdelingView,
)
from core_service import database

db = database.db


def create_app(config_name='huishoudboekje_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    db.init_app(app)
    migrate = Migrate(app, db)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/burgers", "view": BurgerView, "name": "burger_view"},
        {"path": "/burgers/<object_id>", "view": BurgerView, "name": "burger_detail_view"},
        {"path": "/burgers/<object_id>/rekeningen", "view": RekeningBurgerView,
         "name": "burger_rekeningen_view"},
        {"path": "/afspraken", "view": AfspraakView, "name": "afspraak_view"},
        {"path": "/afspraken/<object_id>", "view": AfspraakView, "name": "afspraak_detail_view"},
        {"path": "/rekeningen", "view": RekeningView, "name": "rekening_view"},
        {"path": "/rekeningen/<object_id>", "view": RekeningView, "name": "rekening_detail_view"},
        {"path": "/journaalposten", "view": JournaalpostView, "name": "journaalpost_view"},
        {"path": "/journaalposten/<object_id>", "view": JournaalpostView, "name": "journaalpost_detail_view"},
        {"path": "/rubrieken", "view": RubriekView, "name": "rubriek_view"},
        {"path": "/rubrieken/<object_id>", "view": RubriekView, "name": "rubriek_detail_view"},
        {"path": "/configuratie", "view": ConfiguratieView, "name": "configuratie_view"},
        {"path": "/configuratie/<object_id>", "view": ConfiguratieView, "name": "configuratie_detail_view"},
        {"path": "/overschrijvingen", "view": OverschrijvingView, "name": "overschrijving_view"},
        {"path": "/overschrijvingen/<object_id>", "view": OverschrijvingView, "name": "overschrijving_detail_view"},
        {"path": "/export", "view": ExportView, "name": "export_view"},
        {"path": "/export/<object_id>", "view": ExportView, "name": "export_detail_view"},
        {"path": "/huishoudens", "view": HuishoudenView, "name": "huishouden_view"},
        {"path": "/huishoudens/<object_id>", "view": HuishoudenView, "name": "huishouden_detail_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView, "name": "afdeling_detail_view"},
        {"path": "/afdelingen/<object_id>/rekeningen", "view": RekeningAfdelingView,
         "name": "afdeling_rekeningen_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app
