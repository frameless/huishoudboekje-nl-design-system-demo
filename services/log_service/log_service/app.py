""" Main app for log_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from core_service import database
from log_service.views.gebruikersactiviteit import GebruikersActiviteitView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(
    config_name="log_service.config.Config"
):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel("WARNING")

    add_statsd_metrics(app)

    @app.route("/health")
    def health():
        return Response()

    # Views
    routes = [
        {
            "path": "/gebruikersactiviteiten",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_view",
        },
        {
            "path": "/gebruikersactiviteiten/<object_id>",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_detail_view",
        },
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False,
        )

    return app


if __name__ == "__main__":
    create_app().run()
