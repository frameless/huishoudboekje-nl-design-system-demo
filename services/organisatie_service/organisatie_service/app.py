""" Main app for organisatie_service """
from models import *
import logging

from flask import Flask, Response
from flask_migrate import Migrate

from organisatie_service.views.organisaties import OrganisatieView
from organisatie_service.views.organisatie_rekeningen import OrganisatieRekeningenView
from organisatie_service.views.afdelingen import AfdelingView
from core_service import database
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='organisatie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_sqlalchemy_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    routes = [
        {"path": "/organisaties", "view": OrganisatieView, "name": "organisatie_view"},
        {"path": "/organisaties/<object_id>", "view": OrganisatieView,
            "name": "organisatie_detail_view"},
        {"path": "/organisaties/rekeningen", "view": OrganisatieRekeningenView, "name": "organisatie_rekeningen_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView,
            "name": "afdeling_detail_view"},
    ]

    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
