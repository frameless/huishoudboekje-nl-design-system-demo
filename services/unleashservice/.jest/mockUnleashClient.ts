import {DeepMockProxy, mockDeep, mockReset} from "jest-mock-extended";
import {Unleash} from "unleash-client";
import unleash from "../src/unleash";

jest.mock("../src/unleash", () => ({
	__esModule: true,
	default: mockDeep<Unleash>(),
}));

beforeEach(() => {
	mockReset(unleashMock);
});

export const unleashMock = unleash as unknown as DeepMockProxy<Unleash>;
