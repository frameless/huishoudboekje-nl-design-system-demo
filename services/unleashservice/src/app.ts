import cors from "cors";
import express from "express";
import apiRouter from "./api-v1";
import healthRouter from "./health";
import unleashClient from "./unleash";

const createApp = () => {
	const app = express();
	app.use(cors());

	const features = unleashClient.getFeatureToggleDefinitions();
	console.info("Available features:");
	console.table((features || []).map(f => ({...f, strategies: JSON.stringify(f.strategies)})));

	app.get("/health", healthRouter);
	app.use("/api/unleash", apiRouter);

	return app;
};

export default createApp;
