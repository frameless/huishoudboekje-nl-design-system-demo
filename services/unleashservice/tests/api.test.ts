import supertest from "supertest";
import {FeatureInterface} from "unleash-client/lib/feature";
import {unleashMock} from "../.jest/mockUnleashClient";
import app from "../src/app";

const api = supertest(app());

const featureFlags: FeatureInterface[] = [
	{name: "feature1", description: "Some description1", enabled: true, stale: false, impressionData: false, strategies: [], variants: []},
	{name: "feature2", description: "Some description2", enabled: false, stale: false, impressionData: false, strategies: [], variants: []},
	{name: "feature3", description: "Some description3", enabled: true, stale: false, impressionData: false, strategies: [], variants: []},
];

const mockFeatures = (featureFlags: FeatureInterface[] = []) => {
	unleashMock.getFeatureToggleDefinitions.mockReturnValue(featureFlags);
	unleashMock.isEnabled.mockImplementation((name) => {
		const featureFlag = featureFlags.find(f => f.name === name);
		return !!featureFlag?.enabled;
	});
};

describe("Unleash", () => {

	it("should return the entire list of features", async () => {
		mockFeatures(featureFlags);

		const result = await api.get("/api/unleash");
		expect(result.status).toBe(200);
		expect(unleashMock.getFeatureToggleDefinitions).toHaveBeenCalledTimes(1);
		expect(result.body).toMatchSnapshot();
		expect(result.body.features.length).toStrictEqual(featureFlags.length);
	});

	describe("Given the feature flags exist", () => {
		it("should return true or false for a single feature", async () => {
			mockFeatures(featureFlags);

			const featureFlagName = featureFlags[0].name;

			const result = await api.post("/api/unleash/" + featureFlagName);
			expect(result.status).toBe(200);
			expect(unleashMock.isEnabled).toHaveBeenCalledTimes(1);
			expect(result.body).toMatchSnapshot();
			expect(result.body[featureFlagName]).toBe(true);
		});

		it("should return true or false for multiple features", async () => {
			mockFeatures(featureFlags);

			const result = await api.post("/api/unleash/" + featureFlags.map(f => f.name).join(","));
			expect(unleashMock.isEnabled).toHaveBeenCalledTimes(featureFlags.length);
			expect(result.body).toMatchSnapshot();
			expect(result.status).toBe(200);
		});

		it("should return an empty list for no passed flags", async () => {
			mockFeatures(featureFlags);

			const result = await api.post("/api/unleash/");
			expect(unleashMock.isEnabled).toHaveBeenCalledTimes(0);
			expect(result.body).toMatchSnapshot();
		});
	});

	describe("Given the feature flags don't exist", () => {
		it("should return false for a single feature", async () => {
			mockFeatures([]);

			const featureFlagName = "non-existent-feature";

			const result = await api.post("/api/unleash/" + featureFlagName);
			expect(result.status).toBe(200);
			expect(unleashMock.isEnabled).toHaveBeenCalledTimes(1);
			expect(result.body).toMatchSnapshot();
			expect(result.body[featureFlagName]).toBe(false);
		});

		it("should return false for multiple features", async () => {
			mockFeatures([]);

			const result = await api.post("/api/unleash/" + featureFlags.map(f => f.name).join(","));
			expect(unleashMock.isEnabled).toHaveBeenCalledTimes(featureFlags.length);
			expect(result.status).toBe(200);
			expect(result.body).toMatchSnapshot();
			Object.values(result.body).forEach((f) => {
				expect(f).toBe(false);
			});
		});

	});

	describe("Given the server is healthy", () => {
		it("should return 200 OK", async () => {
			const result = await api.get("/health");
			expect(result.statusCode).toEqual(200);
		});
	});

});