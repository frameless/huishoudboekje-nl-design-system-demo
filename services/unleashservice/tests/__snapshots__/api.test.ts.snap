// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unleash Given the feature flags don't exist should return false for a single feature 1`] = `
{
  "non-existent-feature": false,
}
`;

exports[`Unleash Given the feature flags don't exist should return false for multiple features 1`] = `
{
  "feature1": false,
  "feature2": false,
  "feature3": false,
}
`;

exports[`Unleash Given the feature flags exist should return all the available flags when no flags are passed 1`] = `
{
  "features": [
    {
      "enabled": true,
      "name": "feature1",
    },
    {
      "enabled": false,
      "name": "feature2",
    },
    {
      "enabled": true,
      "name": "feature3",
    },
  ],
}
`;

exports[`Unleash Given the feature flags exist should return true or false for a single feature 1`] = `
{
  "feature1": true,
}
`;

exports[`Unleash Given the feature flags exist should return true or false for multiple features 1`] = `
{
  "feature1": true,
  "feature2": false,
  "feature3": true,
}
`;

exports[`Unleash should return the entire list of features 1`] = `
{
  "features": [
    {
      "enabled": true,
      "name": "feature1",
    },
    {
      "enabled": false,
      "name": "feature2",
    },
    {
      "enabled": true,
      "name": "feature3",
    },
  ],
}
`;
